go/vt/vtgate/executor.go

```diff
@@ -304,11 +302,11 @@ func (e *Executor) handleExec(ctx context.Context, safeSession *SafeSession, sql
 	return qr, err
 }
 
-func (e *Executor) destinationExec(ctx context.Context, safeSession *SafeSession, sql string, bindVars map[string]*querypb.BindVariable, target querypb.Target, destination key.Destination, logStats *LogStats) (*sqltypes.Result, error) {
-	return e.resolver.Execute(ctx, sql, bindVars, target.Keyspace, target.TabletType, destination, safeSession.Session, false /* notInTransaction */, safeSession.Options, logStats)
+func (e *Executor) destinationExec(ctx context.Context, safeSession *SafeSession, sql string, bindVars map[string]*querypb.BindVariable, destTarget key.DestinationTarget, logStats *LogStats) (*sqltypes.Result, error) {
+	return e.resolver.Execute(ctx, sql, bindVars, destTarget.Keyspace, destTarget.TabletType, destTarget.Destination, safeSession.Session, false /* notInTransaction */, safeSession.Options, logStats)
```

Comment:
I feel that the previous one was wrong, it is:
target querypb.Target, destination key.Destination
should have been:
keyspace string, targetType, destination
See my previous comment, we don't really need a structure for this, do we?